{
	"id": "CVE-2022-41909",
	"vulnerabilities": [
		{
			"content": {
				"id": "CVE-2022-41909",
				"description": "TensorFlow is an open source platform for machine learning. An input `encoded` that is not a valid `CompositeTensorVariant` tensor will trigger a segfault in `tf.raw_ops.CompositeTensorVariantToComponents`. We have patched the issue in GitHub commits bf594d08d377dc6a3354d9fdb494b32d45f91971 and 660ce5a89eb6766834bdc303d2ab3902aef99d3d. The fix will be included in TensorFlow 2.11. We will also cherrypick this commit on TensorFlow 2.10.1, 2.9.3, and TensorFlow 2.8.4, as these are also affected and still in supported range.",
				"severity": [
					{
						"type": "cvss_v31",
						"source": "nvd@nist.gov",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
							"base_score": 7.5,
							"base_severity": "HIGH",
							"temporal_score": 7.5,
							"temporal_severity": "HIGH",
							"environmental_score": 7.5,
							"environmental_severity": "HIGH"
						}
					},
					{
						"type": "cvss_v31",
						"source": "security-advisories@github.com",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:H/PR:L/UI:R/S:U/C:N/I:N/A:H",
							"base_score": 4.8,
							"base_severity": "MEDIUM",
							"temporal_score": 4.8,
							"temporal_severity": "MEDIUM",
							"environmental_score": 4.8,
							"environmental_severity": "MEDIUM"
						}
					}
				],
				"cwe": [
					{
						"source": "nvd@nist.gov",
						"cwe": [
							"CWE-476"
						]
					},
					{
						"source": "security-advisories@github.com",
						"cwe": [
							"CWE-20"
						]
					}
				],
				"references": [
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/lib/core/py_func.cc"
					},
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/commit/660ce5a89eb6766834bdc303d2ab3902aef99d3d"
					},
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/commit/bf594d08d377dc6a3354d9fdb494b32d45f91971"
					},
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-rjx6-v474-2ch9"
					},
					{
						"source": "nvd.nist.gov",
						"url": "https://nvd.nist.gov/vuln/detail/CVE-2022-41909"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/lib/core/py_func.cc"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/commit/660ce5a89eb6766834bdc303d2ab3902aef99d3d"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/commit/bf594d08d377dc6a3354d9fdb494b32d45f91971"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-rjx6-v474-2ch9"
					}
				],
				"published": "2022-11-18T22:15:22.223Z",
				"modified": "2024-11-21T07:24:02.807Z"
			},
			"segments": [
				{
					"ecosystem": "cpe"
				}
			]
		}
	],
	"detections": [
		{
			"ecosystem": "cpe",
			"conditions": [
				{
					"criteria": {
						"operator": "OR",
						"criterias": [
							{
								"operator": "OR",
								"criterias": [
									{
										"operator": "OR",
										"criterias": [
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
															},
															"affected": {
																"type": "semver",
																"range": [
																	{
																		"lt": "2.8.4"
																	}
																]
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
															},
															"affected": {
																"type": "semver",
																"range": [
																	{
																		"lt": "2.9.3",
																		"ge": "2.9.0"
																	}
																]
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:2.10.0:*:*:*:*:*:*:*"
															}
														}
													}
												]
											}
										]
									}
								]
							}
						]
					}
				}
			]
		}
	],
	"data_source": {
		"id": "nvd-api-cve",
		"raws": [
			"vuls-data-raw-nvd-api-cve/2022/CVE-2022-41909.json"
		]
	}
}
